src/client/app/components/LoginComponent.tsx

```diff
@@ -94,27 +94,24 @@ export default class LoginComponent extends React.Component<{}, LoginState> {
 	 */
 	private handleSubmit(e: React.FormEvent<HTMLFormElement>) {
 		e.preventDefault();
-		axios.post('/api/login/', {
-			email: this.state.email,
-			password: this.state.password
-		})
-		.then(response => {
-			localStorage.setItem('token', response.data.token);
-			browserHistory.push('/');
-		})
-		.catch(err => {
-			if (err.response.status === 401) {
-				showErrorNotification('Invalid email/password combination');
-			} else {
-				// If there was a problem other than a lack of authorization, the user can't fix it.
-				// Log it to the console for developer use.
-				console.error(err); // tslint:disable-line no-console
+		(async () => {
```

Comment:
Would it be possible to mark `handleSubmit` as async instead of using an async IFFE? Or does that not work with React?

Reply:
I can't call setState() inside an async function, so this has to stay an IFFE.